go/main.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/google/go-github/v68/github"
+)
+
+const (
+	SlsaSourceLevel1 = "SLSA_SOURCE_LEVEL_1"
+	SlsaSourceLevel2 = "SLSA_SOURCE_LEVEL_2"
+)
+
+type activity struct {
+	Id           int
+	Before       string
+	After        string
+	Ref          string
+	Timestamp    time.Time
+	ActivityType string `json:"activity_type"`
+}
+
+// Checks to see if the rule meets our requirements.
+func checkRule(ctx context.Context, gh_client *github.Client, owner string, repo string, rule *github.RepositoryRule, minTime time.Time) (bool, error) {
+	ruleset, _, err := gh_client.Repositories.GetRuleset(ctx, owner, repo, rule.RulesetID, false)
```

Comment:
as mentioned elsewhere, this only tells you what's configured now, if a new revision were to be created right now -- it doesn't say anything about the revision under consideration. 

This kind of logic made sense in previous iterations of the source track where "the entire repo" was marked compliant or not. 
For many good reasons we've tried to move to a revision-based model, where only certain revisions are compliant or not. 

Unless the repo is part of an organization, the rule_suites apis to learn about a specific revision don't really exist yet. you'd have to do some digging to make it work today.

Reply:
Yup, so I improved how this is done here: https://github.com/slsa-framework/slsa-source-poc/blob/main/sourcetool/pkg/checklevel/checklevel.go#L94

There we check
1. The time the rule started being active
2. The time the repo 'opted in to SLSA'
3. The time the commit occurred
4. We check that the commit occurred after the repo opted in to SLSA
5. That the rules were enabled at least as long as the repo was opted in to SLSA

I think the net result is that we cover this case?